app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java

```diff
@@ -266,42 +271,74 @@ public static void showSttContextDialog(Activity activity, List<String> availabl
         dopt.data = new ArrayList<>(new TreeSet<>(availableData));
         dopt.callback = callback;
         dopt.titleText = isTodoTxtAlternativeNaming(activity) ? R.string.category : R.string.context;
-        dopt.searchHintText = R.string.search_or_custom;
-        //dopt.messageText = activity.getString(R.string.add_x_or_browse_existing_ones_witharg, activity.getString(R.string.context));
+        dopt.multiSelectCallback = (result) -> {
+            for (final Integer pi : result) {
+                callback.callback(dopt.data.get(pi).toString());
+            }
+        };
```

Comment:
Add multiple contexts or projects with multi-select

Reply:
so you just fire the callback which does in turn insert/append ... multiple times?

btw what does "pi" mean here?

Reply:
> so you just fire the callback which does in turn insert/append ... multiple times?

Yes. The call back maintains the selection correctly so this works well.

Leaving the variable as `pi` was a mistake. It was originally 'project index' before I combined the project and context functions. Should change it to `di` for data index or something.

Reply:
just i or index is finr

Reply:
changed